### PR TITLE
Add doc for observability

### PR DESCRIPTION
**Pull request by aureamunoz**

### CODE REVIEW

File: docs/docs/observability.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+
+  ![observation_process](target/observation_sequence.svg#only-light)
+  ![observation_process](target/observation_sequence_dark.svg#only-dark)
+
+
+
+## Implementing an observation collector
+
+An `ObservationCollector` implementation needs to override the `create` method in order to provide an instance of StorkObservation.
```
Comment (cescoffier) [inline docs/docs/observability.md:37]:
```suggestion
An `ObservationCollector` implementation must override the `create` method to provide an instance of StorkObservation.
```

File: docs/docs/observability.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+A custom observation collector class should look as follows:
+
+```java linenums="1"
+{{ insert('examples/AcmeObservationCollector.java') }}
+```
+
+The next step is to initialize Stork with an `ObservableStorkInfrastructure` taking an instance of your ObservationCollector as parameter.
```
Comment (cescoffier) [inline docs/docs/observability.md:45]:
```suggestion
The next step is to initialize Stork with an `ObservableStorkInfrastructure`, taking an instance of your `ObservationCollector` as parameter.
```

File: docs/docs/observability.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+```java linenums="1"
+{{ insert('examples/ObservableInitializationExample.java') }}
+```
+
+Then, Stork will use your implementation to register metrics.
```
Comment (cescoffier) [inline docs/docs/observability.md:51]:
```suggestion
Then, Stork uses your implementation to register metrics.
```

File: docs/docs/observability.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+## Observing service discovery and selection behaviours
+
+To access metrics registered by `StorkObservation`, use the following code:
+
+```java linenums="1"
+{{ insert('examples/ObservationExample.java') }}
+```
+
+
```
Comment (cescoffier) [inline docs/docs/observability.md:62]: Too many empty lines.

File: docs/docs/observability.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+# Stork Observability with Quarkus
+
+Stork metrics are automatically enabled when using Stork together with Micrometer extension in a Quarkus application. Micrometer collects the metrics of the Rest or Grpc client using Stork.
```
Comment (cescoffier) [inline docs/docs/observability.md:66]:
```suggestion
Stork metrics are automatically enabled when using Stork together with the Micrometer extension in a Quarkus application. 
Micrometer collects the metrics of the rest clients and grpc clients using Stork, as well as when using the Stork API.
```

File: docs/snippets/examples/AcmeObservationCollector.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package examples;
+
+import io.smallrye.stork.api.observability.ObservationCollector;
+import io.smallrye.stork.api.observability.StorkEventHandler;
+import io.smallrye.stork.api.observability.StorkObservation;
+
+public class AcmeObservationCollector  implements ObservationCollector {
+    private static final StorkEventHandler ACME_HANDLER = ev -> {
+    };
```
Comment (cescoffier) [inline docs/snippets/examples/AcmeObservationCollector.java:9]: Here mention what needs to happen in the implementation.
That's where you get the composite (terminal event)

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/docs/observability.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+The following sequence diagram shows how the observability is initialized : 
+
+
+![observability initialization](target/observability_sequence.svg#only-light)
+![observability initialization](target/observability_sequence_dark.svg#only-dark)
+
+
+
+The `StorkObservation` registers times, discovered instances, the selected instance and failures by reacting to the lifecycle of a Stork event such us:
```
[the remark this answers is not in the document]
Comment (aureamunoz) [reply docs/docs/observability.md]: mmm...yes, I aggree. We can remove it and add it if we feel the need

File: docs/docs/observability.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+- start : Observation has been started. Begin time is registered. Happens when the `ObservationCollector#create()` method gets called.
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/docs/observability.md:23]: Why didn't you integrate this change?

File: docs/docs/observability.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+- service discovery success: a collection of instances has been successfully discovered for a service. End discovery time and number of instances are registered. Happens when the `StorkObservation#onServiceDiscoverySuccess` gets called.
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/docs/observability.md:24]: Why didn't you integrate this change?

File: docs/docs/observability.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+- service discovery error: an error occurred when discovering service. End discovery time and failure cause are registered. Happens when the `StorkObservation#onServiceDiscoveryFailure` gets called.
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/docs/observability.md:25]: Why didn't you integrate this change?

File: docs/docs/observability.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+- service selection success: an instance has been successfully selected from the collection. End selection time and selected instance id are registered. Happens when the `StorkObservation#onServiceSelectionSuccess` gets called.
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/docs/observability.md:26]: Why didn't you integrate this change?

File: docs/docs/observability.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+- service selection error: an error occurred during selecting the instance. End selection time and failure cause are registered. Happens when the `StorkObservation#onServiceSelectionFailure` gets called.
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/docs/observability.md:27]: Why didn't you integrate this change?

File: docs/docs/observability.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+- end: Observation has finished. Overall duration is registered. Happens when the `StorkObservation#onServiceSelectionSuccess` gets called.
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/docs/observability.md:28]: Why didn't you integrate this change?

File: docs/docs/observability.md (hunk 1 above)
Comment (cescoffier) [reply docs/docs/observability.md:40]: Why didn't you integrate this change?

**Review by cescoffier — CHANGES_REQUESTED** [review]
Some of my suggestions from the previous review have not been integrated. Any reason?

**Comment by aureamunoz** [comment]
> Some of my suggestions from the previous review have not been integrated. Any reason?

They were hidden in the UI and I completely missed them. Looking at them now

**Review by cescoffier — APPROVED** [review]
LGTM!

Can you squash and merge?

**Pull request merged**